addons/mail/models/res_company.py

```diff
@@ -8,23 +8,61 @@ class Company(models.Model):
     _name = 'res.company'
     _inherit = 'res.company'
 
+    def _get_default_alias_domain_id(self):
+        return self.env['mail.alias.domain'].search([], limit=1).id
+
+    # alias_domain_ids = fields.One2many('mail.alias.domain', 'company_id', 'Alias Domains')
+    # alias_domain_id = fields.Many2one(
+    #     'mail.alias.domain', 'Alias Domain',
+    #     compute='_compute_alias_domain_id', readonly=False, store=True)
+    alias_domain_id = fields.Many2one(
+        'mail.alias.domain', string='Alias Domain', ondelete='set null',
+        default=lambda self: self._get_default_alias_domain_id())
+    alias_domain_name = fields.Char('Alias Domain Name', related='alias_domain_id.name', readonly=True, store=True)
```

Comment:
To solve the problem explained above, should this be m2m? so we can specify more than 1 domain for each company.

Reply:
After trying various models, it seems the m2m is better suited indeed. There is a "default mail domain" on companies, that will be the default one when having to choose a catchall / bounce in a given company environment, but other domains should be accepted within a given company.